zkevm-circuits/src/keccak_circuit/keccak_padding_byte_2col_lookup.rs

```diff
@@ -0,0 +1,604 @@
+                table.assign_cell(|| "is_padding_col_[i=K+2]", self.is_padding_col, (K + 2) as usize, || Ok(F::from(1)))?;
+
+                //// byte = 1, is_pad = 1, the end of the padding separate from beginning case
+                // self.assign_table_row(table, (K + 3) as usize, 1, 1);
+                table.assign_cell(|| "byte_col_[i=K+3]", self.byte_col, (K + 3) as usize, || Ok(F::from(1)))?;
+                table.assign_cell(|| "is_padding_col_[i=K+3]", self.is_padding_col, (K + 3) as usize, || Ok(F::from(1)))?;
+
+                Ok(())
+            },
+        )
+    }
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////// 
+//////////////////////////////////////////////////////////////////////////////////////////////////////////// 
+//////////////////////////////////////////////////////////////////////////////////////////////////////////// 
+//////////////////////////////////////////////////////////////////////////////////////////////////////////// 
+
+#[derive(Clone, Debug)]
+#[allow(missing_docs)]
+pub struct KeccakPaddingConfig<F> {
+    q_enable: Selector,
+    q_end: Column<Advice>,
+    d_bytes: [Column<Advice>; KECCAK_WIDTH_IN_BYTES],
+    d_lens: [Column<Advice>; KECCAK_RATE_IN_BYTES],
+    d_rlcs: [Column<Advice>; KECCAK_RATE_IN_BYTES],
+    is_pads: [Column<Advice>; KECCAK_RATE_IN_BYTES],
+    randomness: Column<Advice>,
+    _marker: PhantomData<F>,
+    padding_combinations_table: PaddingCombinationsConfig<F>,
+}
+
+impl<F: Field> KeccakPaddingConfig<F> {
+    pub(crate) fn configure(meta: &mut ConstraintSystem<F>) -> Self {
+        let randomness = meta.advice_column();        
+        let q_enable = meta.selector();
+        let q_end = meta.advice_column();
+
+        let padding_combinations_table = PaddingCombinationsConfig::<F>::configure(meta);
+
+        let d_bytes = [(); KECCAK_WIDTH_IN_BYTES].map(|_| meta.advice_column());
+        let d_lens = [(); KECCAK_RATE_IN_BYTES].map(|_| meta.advice_column());
+        let d_rlcs = [(); KECCAK_RATE_IN_BYTES].map(|_| meta.advice_column());
+        let is_pads = [(); KECCAK_RATE_IN_BYTES].map(|_| meta.advice_column());
+        
+        // Check that (padding flag, data/padding byte) combinations are restricted to the combinations in padding_combinations_table.        
+        // The table will estrict padding flags to {0,1} and data/padding bytes to [0, 255].
+        // Only particular combinations are allowed, check PaddingCombinationsConfig load method comments.
+        for i in 0..is_pads.len() {
+            meta.lookup("Check allowed data/padding/flag combinations", |meta| {
+                // let q_enable = meta.query_selector(q_enable);
+                vec![
+                    (meta.query_advice(d_bytes[i], Rotation::cur()), padding_combinations_table.byte_col),
+                    (meta.query_advice(is_pads[i], Rotation::cur()), padding_combinations_table.is_padding_col),
```

Comment:
do these 2 lines mean
$d\\_bytes \subset padding\\_combinations\\_table.byte\\_col$ and $is\\_pads \subset padding\\_combinations\\_table.is\\_padding\\_col$
or
$(d\\_bytes, is\\_pads) \subset padding\\_combinations_table.(byte\\_col, is\\_padding\\_col)$

The diff is if the value in table are located in the same row.

Reply:
The last one, both values need to be on the same row in the corresponding table. Internally lookups are reduced to a single value set check.